src/test/unit/ntypes/ValueMapTest.cpp

```diff
@@ -0,0 +1,116 @@
+/* ---------------------------------------------------------------------
+ * HTM Community Edition of NuPIC
+ * Copyright (C) 2013, Numenta, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * --------------------------------------------------------------------- */
+
+/** @file
+ * Implementation of YAML tests
+ */
+
+#include "gtest/gtest.h"
+#include <htm/engine/Network.hpp>
+#include <htm/engine/Spec.hpp>
+#include <htm/ntypes/ValueMap.hpp>
+
+namespace testing { 
+    
+using namespace htm;
+
+TEST(ValueMapTest, IntScalerTest) {
+  ValueMap vm;
+
+  const char *s1 = "10";
+  vm.parse(s1);
+  EXPECT_TRUE(vm.isScalar());
+  UInt32 u = vm.as<UInt32>();
+  EXPECT_EQ(10u, u);
+
+  vm.parse("-1"); 
+  EXPECT_ANY_THROW(UInt32 x = vm.as<UInt32>());  // Unsigned must be positive.
+  Int32 i = vm.as<Int32>();
+  EXPECT_EQ(-1, i);
+
+  vm.parse("- 1");
+  EXPECT_TRUE(vm.isSequence());
+  u = vm[0].as<UInt32>();
```

Comment:
shouldn't "- 1" just fail? and why "1" is in [0], then "-" is then in [1]..parser working right-to-left?

Reply:
> shouldn't "- 1" just fail? and why "1" is in [0], then "-" is then in [1]..parser working right-to-left?

In YAML, a leading "- " (a dash followed by a space) means this is an element of a sequence.
The construct vm[0] means give me the first element of the sequence which should be the scaler "1".